test/controllers/AddClosureMetadataControllerSpec.scala

```diff
@@ -311,6 +312,56 @@ class AddClosureMetadataControllerSpec extends FrontEndTestHelper {
       formTester.checkHtmlForOptionAndItsAttributes(addClosureMetadataPageAsString, formSubmission.toMap, formStatus = "PartiallySubmitted")
     }
 
+    "display an error when future date is not allowed for a date field" in {
+      val consignmentId = UUID.fromString("c2efd3e6-6664-4582-8c28-dcf891f60e68")
+      val addClosureMetadataController = instantiateAddClosureMetadataController()
+      val formTester = new FormTester(expectedDefaultOptions)
+      setConsignmentTypeResponse(wiremockServer, "standard")
+      mockGraphqlResponse()
+      setConsignmentReferenceResponse(wiremockServer)
+      setConsignmentFilesMetadataResponse(wiremockServer)
+
+      val year = LocalDateTime.now().plusYears(1).getYear.toString
+      val formSubmission = Seq(
+        ("inputdate-FoiExemptionAsserted-day", "5"),
+        ("inputdate-FoiExemptionAsserted-month", "11"),
+        ("inputdate-FoiExemptionAsserted-year", year),
+        ("inputdate-ClosureStartDate-day", "1"),
+        ("inputdate-ClosureStartDate-month", "1"),
+        ("inputdate-ClosureStartDate-year", "2022"),
+        ("inputnumeric-ClosurePeriod-years", "0"),
+        ("inputdropdown-FoiExemptionCode", "mock code1"),
+        ("inputradio-TitlePublic", "yes")
+      )
+
+      val addClosureMetadataPage = addClosureMetadataController.addClosureMetadataSubmit(consignmentId)
+        .apply(FakeRequest(POST, s"/standard/$consignmentId/add-closure-metadata")
+          .withFormUrlEncodedBody(formSubmission: _*)
+          .withCSRFToken
+        )
+      val addClosureMetadataPageAsString = contentAsString(addClosureMetadataPage)
+
+      checkPageForStaticElements.checkContentOfPagesThatUseMainScala(addClosureMetadataPageAsString, userType = "standard")
+      checkForExpectedClosureMetadataFormPageContent(addClosureMetadataPageAsString)
+      formTester.checkHtmlForOptionAndItsAttributes(addClosureMetadataPageAsString, formSubmission.toMap, formStatus = "PartiallySubmitted")
+
+      addClosureMetadataPageAsString must include(
```

Comment:
The test is specific to test the error related to future date on the page. So, we cannot add in that method otherwise all other tests will fail.

Reply:
Ah yes, sorry, you're correct. But then, do we need this whole unit test since we are already testing that the correct error message is being returned in the FormFieldSpec? These controller tests are just mainly to test if the controller did the right thing i.e. rendered a page that is displaying/hiding the correct elements (when you GET, POST incorrectly, POST correctly) rather than the specific error messages as these come from FormFields. As long as FormFields tests pass, it should always display the correct text since the views are all calling `fieldErrors`. This is why the error messages I put in `MockInputOption` are only the default ones because these are hidden when you GET but are unhidden when you incorrectly POST.

Reply:
Yeah, you are right. It's not required actually. I'll delete it.